### ISSUE TITLE
Running python chunk interactively in html_notebook hides following R chunk

### ISSUE THREAD

**Issue by cderv** [rstudio/rstudio#8263]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.4.997
    OS Version      :  Windows 10
    R Version       : 4.0.2

### Steps to reproduce the problem

* Open a new Rmd document with this content
````md
---
title: "R Notebook"
output: html_notebook
---

```{r}
library(reticulate)
use_condaenv('r-reticulate')
```

```{python}
import pandas as pd
df = pd.DataFrame({
   "Name": ["Braund, Mr. Owen Harris",
            "Allen, Mr. William Henry",
            "Bonnell, Miss. Elizabeth"],
   "Age": [22, 35, 58],
   "Sex": ["male", "male", "female"]}
   )
df
```

```{r}
py$df
```
````
* Configure **reticulate** to have a `r-reticulate` conda env and `pandas` installed
* When you preview the first time, there is no result
* Run interactively with the green arrow the first two chunks, and preview the doc. All the chunks are present, the last one does not have any result yet.
* First odd thing: The console is still open in a Python REPL - It may be expected though... 
* Run the last R chunk, it will run `quit` in the console, closing the python REPL, another `quit` in the R session, causing printing of the function `quit()` (see screenshots) - and `py$df` will be run in the console. **No output in the Rmd document inline**
* Preview the doc: There will be not last R chunk in the document - it dissapeared.
* Re-rerun the last chunk interactively: Result will be in the Rmd document. Preview the doc and everything is fine. 

### Describe the problem in detail

![notebook-reticulate](https://user-images.githubusercontent.com/6791940/97875878-4f7f5700-1d1b-11eb-9bee-4ae77684f3c8.gif)

Main issue 
* The last R chunk has disappeared from the html notebook after being executed. It needs to be run twice to get incorporated. 

Maybe issues: 
* Python REPL is still open after running a python chunk
* `quit` is sent twice to the console

It could be an issue with reticulate only, or `html_notebook`, or both. But I believe this is something due to integration within the IDE.

### Describe the behavior you expected

I would expect the experience with Python chunk using **reticulate** to be seamless and not require extra step compared to how it works with R usually. 

Am I expecting right ?

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting a RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by kevinushey** [comment]
Marking as 1.4 so I can try to investigate a bit further but may need to punt to patch release depending on how complicated the underlying issue is.

**Comment by mikebessuille** [comment]
Agree, we should investigate this for 1.4.  If it turns out to be risky we can revisit.

**Comment by kevinushey** [comment]
Strangely, I'm not able to reproduce. :-/

Any other ideas what might be going on? The fact that the issue resolves after re-running the chunk makes me think that it's okay to leave this out of 1.4.

**Comment by cderv** [comment]
I cannot reproduce this with the fresh installa of last 1.4.1002 - Odd. May have been a configuration that would have been modified maybe ? Anyway, not important now that it works. Will see if this arises again. 

There is still the duplicated `quit` call though - unexpected in the R console, but unharmful 
````r
> library(reticulate)
> use_condaenv('r-reticulate')
> reticulate::repl_python()
Python 3.8.6 (C:/Users/chris/AppData/Local/r-miniconda/envs/r-reticulate/python.exe)
Reticulate 1.18.9000 REPL -- A Python interpreter in R.
>>> import pandas as pd
>>> df = pd.DataFrame({
...    "Name": ["Braund, Mr. Owen Harris",
...             "Allen, Mr. William Henry",
...             "Bonnell, Miss. Elizabeth"],
...    "Age": [22, 35, 58],
...    "Sex": ["male", "male", "female"]}
...    )
>>> df
                       Name  Age     Sex
0   Braund, Mr. Owen Harris   22    male
1  Allen, Mr. William Henry   35    male
2  Bonnell, Miss. Elizabeth   58  female
>>> quit
> quit
function (save = "default", status = 0, runLast = TRUE) 
.Internal(quit(save, status, runLast))
<bytecode: 0x0000027561521590>
<environment: namespace:base>
> py$df
                      Name Age    Sex
1  Braund, Mr. Owen Harris  22   male
2 Allen, Mr. William Henry  35   male
3 Bonnell, Miss. Elizabeth  58 female
```` 

Is this consider something to fix ? If not I let you close this. 

Sorry for the noise about a non-issue. :-/

**Comment by kevinushey** [comment]
No worries! We hugely appreciate the time you spend finding and documenting these sorts of issues. It's entirely possible that this issue occurs only intermittently, or depends on some other unknown state so it's good to have logged.

We'll keep the issue around in case you (or someone else) bumps into it again later.

Re: the duplicated `quit` calls -- if you are able to find a reproducible example, would you be willing to file an issue?

**Comment by cderv** [comment]
> Re: the duplicated quit calls -- if you are able to find a reproducible example, would you be willing to file an issue?

I can reproduce following the step here, but I'll get the example reduce and open a new one.

**Comment by cderv** [comment]
@kevinushey Still cannot reproduce with 1.4.1006 and the `quit` being sent twice to console has also disappear now with this version... 🤷‍♂️

**Comment by kevinushey** [comment]
Maybe we just accidentally fixed it somehow 😄
